provider/oci/environ.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package oci
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/version"
+
+	"github.com/juju/juju/constraints"
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/environs/config"
+	"github.com/juju/juju/environs/instances"
+	"github.com/juju/juju/instance"
+	"github.com/juju/juju/provider/common"
+	"github.com/juju/juju/storage"
+)
+
+type Environ struct{}
+
+var _ common.ZonedEnviron = (*Environ)(nil)
+var _ storage.ProviderRegistry = (*Environ)(nil)
+var _ environs.Environ = (*Environ)(nil)
+var _ environs.Firewaller = (*Environ)(nil)
+var _ environs.Networking = (*Environ)(nil)
+var _ environs.NetworkingEnviron = (*Environ)(nil)
+
+// AvailabilityZones is defined in the common.ZonedEnviron interface
+func (e *Environ) AvailabilityZones() ([]common.AvailabilityZone, error) {
+	return nil, nil
```

Comment:
My general recommendation is to return errors.NotImplementedf("METHODNAME") for placeholder code. That way it doesn't get left behind by accident.

Reply:
I agree with this.